### PR TITLE
Add withdraw to kamino-sdk

### PR DESCRIPTION
**Pull request by peroxy**
- also updated the codegen, that's why there are so many files changed..
- the important part is in Kamino.ts and tokenutils.ts

### CODE REVIEW

File: packages/kamino-sdk/tests/kamino.test.ts [hunk 1]
```diff
@@ -69,4 +71,25 @@ describe('Scope SDK Tests', () => {
   //   const kamino = new Kamino(cluster, connection);
   //   console.log(await kamino.getWhirlpools([]));
   // });
+
+  // test('should withdraw shares from a strategy', async () => {
```
Comment (silviutroscot) [inline packages/kamino-sdk/tests/kamino.test.ts:75]: why are these tests commented?

File: packages/kamino-sdk/src/whirpools-client/accounts/Whirlpool.ts [hunk 2]
```diff
@@ -1,154 +1,164 @@
-import { PublicKey, Connection } from '@solana/web3.js';
-import BN from 'bn.js'; // eslint-disable-line @typescript-eslint/no-unused-vars
-import * as borsh from '@project-serum/borsh'; // eslint-disable-line @typescript-eslint/no-unused-vars
-import * as types from '../types'; // eslint-disable-line @typescript-eslint/no-unused-vars
-import { PROGRAM_ID } from '../programId';
+import { PublicKey, Connection } from "@solana/web3.js"
```
Comment (silviutroscot) [inline packages/kamino-sdk/src/whirpools-client/accounts/Whirlpool.ts:1]: is this file autogenerated?

**Review by elliotkennedy — APPROVED** [review]
This is good + works - but it looks very restricted - I'm trying to work out who we are catering for with the SDK - could you give a bit more explaination about why we need it over the generated code?

- Takes Keypair for parameter - can't be used in FE because not wallet object
- Cannot control commitment level
- Sets compute budget regardless of whether the extra budget is needed
- Does not allow adding extra ixs to the tx

When writing an SDK I would probably keep to the bare minimum - essentially just constructing the instruction (perhaps with decimal to BN conversion) - but that might be painful for the use-case we are catering for here

**Review by elliotkennedy — APPROVED** [review]
Nice work!

**Pull request merged**